Botany Balance Pass

Sun-Soaked opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
First came the content, now comes the hammer.

- Nukes Megaseed servitors from orbit. 
- Plants now age much, much slower and produce half as quickly.
Ruins that had them now have a ruined seed vendor that can be salvaged for random seeds(and danger).
Ships that had one now have a crate with some thematic starting seeds, and a Strange Seed.
Ghostrole Ruins that relied on having all seeds locally now have a special biogenerator variant that can print a random seed for biomass.

- Adds Genesis Serum. This can be splashed on a tile to make natural grass and some flora. Green your ship!
Genesis Serum was made a while ago, on request for a way to add natural grass and flora to your ship. Since I had it lying around fully coded, I thought I might as well pr it with botany changes.

- Gatfruit found in the seed vault have been replaced with Strange Seeds.

- The chance to get Gatfruit from a demonic portal(plant variety) has dropped from 15% to 5%.

- Corpse flowers now have liquid gibs and formaldehyde again. 

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game
Okay, hear me out

With this and Gardens, botany ships go from a "sit in your vessel for 2 hours" experience to an "explore and forage" one that better fits our feature arc. It goes without saying that this **shouldn't be merged till Overmap 4.2 is**, since it facilitates getting seeds from planets as part of exploration.

Gatfruit are funny, but it takes exactly one seed getting into the hands of a ship with a dna manipulator and the weapon balance is eradicated from the game completely(for the round, at least.) 
This is more problematic here then it was on TG, since our rounds tend to be 5 hours long rather then 1.
This has been long coming. I'll reverse this if we ever get that Plantlock variant we wanted a while ago.

Corpse flowers even have formaldehyde and gibs on tg, not sure what happened there.
<!-- Please add a short description of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding. -->

## Changelog

:cl: 
add: Ruined megaseed servitors can now be found on the frontier, carrying a bounty of seeds for intrepid adventurers.
balance: the time it takes for plants to reach a lethal age has been increased massively.
balance: Plant production time increased a bit to compensate.
balance: megaseed servitors have been removed from ships and ruins. Ships that carried one now have a crate with some starting seeds. 
balance: removes gatfruit from the seed vault pool.
balance: reduces the chance of getting gatfruit from a plant-themed demonic portal significantly.
balance: corpse flowers once again have formaldehyde and liquid gibs.
add: Adds Genesis Serum, a reagent that transforms tiles into natural grass on splash, then causes some natural flora objects to grow. Turn your ship green!
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

Sun-Soaked committed:
the sacrament...

gatfru'nt

Sun-Soaked committed:
makes the seedvend salvage more dangerous

Sun-Soaked commented:
two notes: 
If anyone would be willing to make a custom sprite for the ruined megaseed, I'd appreciate it.
I maaaybe should have pr'd genesis as part of Overmap 4.2, I'll wait to see if zephyr gets mad at me about it

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into notany

ArdentSpark commented:
The selection of seeds that have been mapped in is incredibly poor. I understand not wanting every ship to have 3 seeds of every base plant to have ever existed but this is painful
![image](https://user-images.githubusercontent.com/82124989/222750442-bed13541-6623-41a7-bb31-03a5a4094d22.png)
(This is the seed vault by the way, the ghost role that is physically unable to leave their vault to forage plants)
I feel like we can be more forgiving than this, while still incentivising players to explore and forage from gardens

Sun-Soaked commented:
ardent, the ghostrole spawners have a special variant of biogen that can generate randomly-selected seeds in exchange for biomass. 
Check the "biocultivator" stuff in the changes.

This is the compromise for the fact that these roles can't go around and forage.

ArdentSpark commented:
The biogenerator is great, and totally forgor. 

Still doesn't change the fact that all the botany ships get the same selection of three seeds + one replica pod + one strange seed (Replica Pod + Strange seed is a really nice forced selection, mind) as we've gone from extreme feast of how many seeds we have to extreme famine. It's really nice that seeds are sustainable now, but it takes a lot, and I mean a lot of time to get those seeds. Each garden yields one seed, has their own do-after, has a fair amount of time before you can repick it, and [half the planets don't have them.](Wrong, it's just sand and lava that don't)

I think Gardens are fantasic, I think incentivising garden exploration for more seeds, but I think it would be good to give botany ships a little taste of botany, without allowing them to min max. I think there's a better middle ground than what has been opted for here.

ghost commented:
It's agonizing to try get ingredients for cooking with like this, it'd be interesting if gardens were weighted but gave more plants, so that powerful plants can remain rare while making it easy to forage for basic stuff that you just plan to cook with.

PigeonLord commented:
I'll mostly repeat what everyone else is saying - if you don't want to make _every_ seed in the Servitor available at the start, fine. I don't agree with it, because finding other seeds on planets is still _incredibly_  RNG reliant, and forces people to do stuff not because it's fun but because they have to in order to do what they find fun from the game, but fine.
(On this - one can claim it's like engineering, where you have to find stuff from planets, but you can find basic materials from pretty much every planet, while you have to get incredibly lucky to get seeds.)
 Maybe some seeds should be rare. But you should at least have the essentials mapped in instead of some random garbage - a couple of each of tomatoes, potatoes, apples, white beets, and cabbage, and maybe a fruit (apples?) would be fine.
What I think would be a good starting selection
-Tomato
-Potato
-Cannabis
-Pumpkin
-Apple
-Cabbage
-White beet
-Wheat
-Cotton

Bokkiewokkie commented on code/game/objects/effects/spawners/lootdrop.dm:
```diff
@@ -689,3 +689,107 @@ GLOBAL_LIST_INIT(ws_survivor_default_loot, list(
+		/obj/item/seeds/cherry/bomb,
+		/obj/item/seeds/berry/glow,
+		/obj/item/seeds/sunflower/moonflower)
+	lootcount = 1
```
Technically not needed to have this, 1 is already the default lootcount. The same applies to the other lootdrops

Bokkiewokkie commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -1801,6 +1801,38 @@
+	var/ground_type = /turf/open/floor/grass
+
+/datum/reagent/genesis/expose_turf(turf/T, reac_volume)
+	if(istype(T, /turf/open/floor/grass))//prevents spamming effect via. smoke or such
```
```suggestion
	if(istype(T, ground_type))//prevents spamming effect via. smoke or such
```
I'd either replace this or remove ground_type entirely and just put grass in the other reference to it.

Sun-Soaked replied:
yeagh. Comitting now

Sun-Soaked committed:
req change 1

Sun-Soaked committed:
silly behaviors

MarkSuckerberg reviewed (approved):
Looks good codewise, waiting until Overmap 4.2 and tweaks for merge though :+1:

TheLeopold commented:
You don't seem to understand that people buy botany ships to 'sit in your vessel for 2 hours', that's just what botany is. 
This turns it into 'pray to rngesus for seed you want, so you can finally to sit in your vessel for 2 hours'. Thus adding even more preparations required to start.
And no, I don't have any counter-suggestions how to do it properly. Leave it alone. Plants grow slowly, it's their nature. You can't change how people interact with botany without changing the core aspects of botany.

Sun-Soaked committed:
Merge branch 'shiptest-ss13:master' into notany

Sun-Soaked commented:
People are always gonna be attached to gameplay they've had fun with in the past. I played an unhealthy amount of pre-2017 nerfs tg botany, and I complained viciously against all the changes that nerfed it.
I still feel like it wasn't the right move, but in many ways I now understand.

Botany in its current form is a powerleveling system that accepts time and busywork in exchange for guaranteed char empowerment. 
This is bad for a lot of reasons.

Guaranteeing access to powerful combat capabilities or the ability to print massive amounts of highest tier medicine cheapens those things, and ultimately cheapens the game experience.
As we work more into extending salvage, scarcity, and the economy, this'll get even more important.
The infinite medicine/weapon glitch(no virus)(30 minutes needed) is even more impactful here than it was on tg, since rounds last pretty long and the time payment is no longer a limiting factor.

Making it less reliable and less reliant on just. having everything right away. is kinda necessary to bring it into line with our other existing gameplay systems(at least in theory, once necro spires and portals are given a sanity check) and our future plans for scavenging, exploration, and the economy.

Sun-Soaked commented:
I may make plants age more slowly and further mess around with tray upkeep stuff in a later pr- To make it into more of a thing you can leave cooking while going off to forage for more seeds.

Sun-Soaked committed:
whoa! Idle farming sim

slows the rate of plants reaching old age by a factor of 20. Babysitting time is OVER

halves the speed of plant production

Sun-Soaked committed:
AIEEEE

perhaps I treated you too harshly
